object_store/src/local.rs

```diff
@@ -678,6 +618,94 @@ impl ObjectStore for LocalFileSystem {
     }
 }
 
+impl LocalFileSystem {
+
+    fn list_with_maybe_offset(&self, prefix: Option<&Path>, maybe_offset: Option<&Path>) -> BoxStream<'static, Result<ObjectMeta>> {
+
+        let config = Arc::clone(&self.config);
+
+        let root_path = match prefix {
+            Some(prefix) => match config.prefix_to_filesystem(prefix) {
+                Ok(path) => path,
+                Err(e) => return futures::future::ready(Err(e)).into_stream().boxed(),
+            },
+            None => config.root.to_file_path().unwrap(),
+        };
+
+        let walkdir = WalkDir::new(root_path)
+            // Don't include the root directory itself
+            .min_depth(1)
+            .follow_links(true);
+
+        let maybe_offset_clone = maybe_offset.map(move |o| o.clone());
+
+        let s = walkdir.into_iter().flat_map(move |result_dir_entry| {
+            // Apply offset filter before proceeding, to reduce statx file system calls
+            // This matters for NFS mounts
+            if maybe_offset_clone.as_ref().is_some() && result_dir_entry.is_ok() {
```

Comment:
Rather than needing the unwrap calls you should be able to do something like:
```rust
if let (Some(offset), Ok(entry)) = (maybe_offset.as_ref(), result_dir_entry) {
```

Reply:
Sounds good. Applied with tweak.